### ISSUE TITLE
stri_trans_general(str, id = "Latin-ASCII") behaves different on different OSs (and maybe locales)

### ISSUE THREAD

**Issue by Tazinho** [gagolews/stringi#305]
Basically see here: https://github.com/sfirke/janitor/issues/204

Me and other users have experienced different transliterations for the letter 'δ' depending on OSs, but also differences on the same OS (maybe because of different locale settings).

Version 1 (on windows and mac):
```
> stringi::stri_trans_general(δ, id = "Latin-ASCII")
[1] "δ"
```

Version 2 (on windows):
```
> stringi::stri_trans_general(δ, id = "Latin-ASCII")
[1] "d"
```

**Comment by gagolews** [comment]
are they using the same version of stringi and the same icu? see stri_info()

**Comment by Tazinho** [comment]
Before I forget: What would be the correct output? I'd guess converting to `d` ?

I am not sure, which output helps you the most. So ... here for Windows 10 on a fresh packrat project with the CRAN versions:

Copy pasted the delta into the code
```
> stringi::stri_trans_general("δ", id = "Latin-ASCII")
[1] "d"
```

Read in external data via **readxl** (see the reprex example at the janitor issue)

```
> stringi::stri_trans_general(names(dat), id = "Latin-ASCII")
[1] "Sample ID" "δ2H, in ‰"
```

Here the output of `stri_info()`

> stringi::stri_info()
$Unicode.version
[1] "7.0"

$ICU.version
[1] "55.1"

$Locale
$Locale$Language
[1] "de"

$Locale$Country
[1] "DE"

$Locale$Variant
[1] ""

$Locale$Name
[1] "de_DE"


$Charset.internal
[1] "UTF-8"  "UTF-16"

$Charset.native
$Charset.native$Name.friendly
[1] "windows-1252"

$Charset.native$Name.ICU
[1] "ibm-5348_P100-1997"

$Charset.native$Name.UTR22
[1] "ibm-5348_P100-1997"

$Charset.native$Name.IBM
[1] "ibm-5348"

$Charset.native$Name.WINDOWS
[1] "windows-1252"

$Charset.native$Name.JAVA
[1] "windows-1252"

$Charset.native$Name.IANA
[1] "windows-1252"

$Charset.native$Name.MIME
[1] NA

$Charset.native$ASCII.subset
[1] TRUE

$Charset.native$Unicode.1to1
[1] TRUE

$Charset.native$CharSize.8bit
[1] TRUE

$Charset.native$CharSize.min
[1] 1

$Charset.native$CharSize.max
[1] 1


$ICU.system
[1] FALSE
```

**Comment by Tazinho** [comment]
@seyounger and @sfirke can you add yours?

**Comment by gagolews** [comment]
Yup, that should be **d**.

**Comment by gagolews** [comment]
Also, could you `str_enc_toutf32(that_δ2H, in ‰_input_string)`?

**Comment by gagolews** [comment]
maybe the string is denormalized, also try stri_trans_tonfd and then stri_trans_tonfc before the Latin-ASCII thing

**Comment by Tazinho** [comment]
```
> input_string
[1] "δ2H, in ‰"
> stringi::stri_enc_toutf32(input_string)
[[1]]
[1]  948   50   72   44   32  105  110   32 8240

> stringi::stri_enc_toutf32("δ2H, in ‰")
[[1]]
[1]  100   50   72   44   32  105  110   32 8240
```

**Comment by gagolews** [comment]
Hmmm... well, on the other hand, this is Latin-ASCII, and not Greek-ASCII... maybe this delta should be kept as-is...

I get:

```
> stringi::stri_trans_general("δ", "Latin-ASCII")
[1] "δ"
> stringi::stri_info(TRUE)
[1] "stringi_1.2.3 (pl_PL.UTF-8; ICU4C 61.1 [system]; Unicode 10.0)"

```

**Comment by gagolews** [comment]
And:

```
> stringi::stri_trans_general("δ", id = "Greek-Latin")
[1] "d"
```

**Comment by gagolews** [comment]
what about:

```
> stri_sort(stringi::stri_trans_general("→δąść©", id = "Greek-Latin; Latin-ASCII"))
[1] "→dasc(C)"
```

**Comment by Tazinho** [comment]
Hmmm, first the output you asked for (doesn't work):

```
> stringi::stri_trans_nfd(input_string)
[1] "δ2H, in ‰"
> stringi::stri_trans_general(
+ stringi::stri_trans_nfd(input_string),
+ id = "Latin-ASCII"
+ )
[1] "δ2H, in ‰"
> stringi::stri_trans_nfc(input_string)
[1] "δ2H, in ‰"
> stringi::stri_trans_general(
+ stringi::stri_trans_nfc(input_string),
+ id = "Latin-ASCII"
+ )
[1] "δ2H, in ‰"
```

**Comment by gagolews** [comment]
i.e., the operations may be chained

**Comment by Tazinho** [comment]
```
> stringi::stri_sort(stringi::stri_trans_general("→δąść©", id = "Greek-Latin; Latin-ASCII"))
[1] "→dasc(C)"
```

**Comment by Tazinho** [comment]
But copy pasting special symbols into R command line might be not the best idea.

**Comment by Tazinho** [comment]
So if you say xxx-ASCII will only work in special cases, it might be good to create some wrapper case anyting-ASCII on the snakecase/janitor side.

Also I am still curious why the other guys get the transliteration into **d**.

**Comment by gagolews** [comment]
Me too, wonder what icu do they use, I mean, this is on the icu, not the stringi side (perhaps)

**Comment by seyounger** [comment]
My output from stri_info()

> stringi::stri_info()
$Unicode.version
[1] "10.0"

$ICU.version
[1] "61.1"

$Locale
$Locale$Language
[1] "en"

$Locale$Country
[1] "US"

$Locale$Variant
[1] ""

$Locale$Name
[1] "en_US"


$Charset.internal
[1] "UTF-8"  "UTF-16"

$Charset.native
$Charset.native$Name.friendly
[1] "UTF-8"

$Charset.native$Name.ICU
[1] "UTF-8"

$Charset.native$Name.UTR22
[1] NA

$Charset.native$Name.IBM
[1] "ibm-1208"

$Charset.native$Name.WINDOWS
[1] "windows-65001"

$Charset.native$Name.JAVA
[1] "UTF-8"

$Charset.native$Name.IANA
[1] "UTF-8"

$Charset.native$Name.MIME
[1] "UTF-8"

$Charset.native$ASCII.subset
[1] TRUE

$Charset.native$Unicode.1to1
[1] NA

$Charset.native$CharSize.8bit
[1] FALSE

$Charset.native$CharSize.min
[1] 1

$Charset.native$CharSize.max
[1] 3


$ICU.system
[1] FALSE

**Comment by gagolews** [comment]
possibly related to #314 (?)

**Comment by gagolews** [comment]
(thread inactive for >= 12 months; closing)

(note that as per #401 stringi is now shipped with ICU 69.1, so the above might be fixed now)
